terraform/010-cluster/main.tf

```diff
@@ -46,6 +68,25 @@ module "asg" {
   tags                    = var.tags
 }
 
+/*
+ * Create secondary auto-scaling group
+ */
+module "asg_secondary" {
+  count = var.aws_region_secondary == "" ? 0 : 1
+
+  source                  = "github.com/silinternational/terraform-modules//aws/asg?ref=8.1.0"
+  app_name                = var.app_name
+  app_env                 = var.app_env
+  aws_instance            = var.aws_instance
+  private_subnet_ids      = module.vpc_secondary.private_subnet_ids
+  default_sg_id           = module.vpc_secondary.vpc_default_sg_id
+  ecs_instance_profile_id = var.ecs_instance_profile_id
+  ecs_cluster_name        = var.ecs_cluster_name
+  ami_id                  = data.aws_ami.ecs_ami.id
+  additional_user_data    = var.asg_additional_user_data
+  tags                    = var.tags
+}
+
```

Comment:
The asg name is based only on `app_name` and `app_env` so will not be unique across zones. I think this is not a problem, only because I could not find anything that said it was a problem -- at least for ASG and a couple other resources I researched. But I wonder if we should go through `terraform-modules` and make all the names include the region as well, just to be safe. Unfortunately, that has backward compatibility issues that need to be considered, so maybe we just march forward and see what breaks.

Reply:
I would suggest we only add region to the name where AWS requires us to (only fixing it when AWS APIs reject attempts to do without), both to minimize work and to avoid making names longer unnecessarily (and to hopefully reduce occurences of backwards compatibility issues that you brought up).